proparse/src/main/java/org/prorefactor/treeparser/ParseUnit.java

```diff
@@ -153,6 +155,26 @@ public void lexAndGenerateMetrics() throws RefactorException {
     }
   }
 
+  /**
+   * Equivalent to PREPROCESS option of COMPILE statement
+   */
+  public void preprocess(Writer writer) throws IOException {
+    ProgressLexer lexer = new ProgressLexer(session, file.getPath(), fileNameList, false);
+    TokenStream stream = lexer.getANTLR2TokenStream(false);
+    try {
+      Token tok = stream.nextToken();
+      while (tok.getType() != Token.EOF_TYPE) {
+        if ((tok.getType() != ProParserTokenTypes.AMPSCOPEDDEFINE)
+            && (tok.getType() != ProParserTokenTypes.AMPGLOBALDEFINE)) {
+          writer.write(tok.getText());
+        }
+        tok = stream.nextToken();
+      }
+    } catch (TokenStreamException uncaught) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS00108)